applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_local_stress_response_function.cpp

```diff
@@ -0,0 +1,336 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder 
+//   
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "adjoint_local_stress_response_function.h"
+
+namespace Kratos
+{
+    /// Default constructor.
+    AdjointLocalStressResponseFunction::AdjointLocalStressResponseFunction(ModelPart& rModelPart, Parameters ResponseSettings)
+    : AdjointStructuralResponseFunction(rModelPart, ResponseSettings)
+    {
+        ModelPart& r_model_part = this->GetModelPart();
+
+        ResponseData stress_response_data;
+
+        // Get traced element
+        mIdOfTracedElement = ResponseSettings["traced_element_id"].GetInt();
+        mpTracedElement = r_model_part.pGetElement(mIdOfTracedElement);
+
+        // Tell traced element the stress type
+        TracedStressType traced_stress_type = stress_response_data.ConvertStressType(ResponseSettings["stress_type"].GetString()); 
+        KRATOS_ERROR_IF(traced_stress_type == TracedStressType::StressTypeNotAvailable) << "Chosen stress type is not available!" << std::endl;
+        mpTracedElement->SetValue(TRACED_STRESS_TYPE, static_cast<int>(traced_stress_type) );		
```

Comment:
why is casting needed here?

Reply:
```TRACED_STRESS_TYPE``` is a variable of type Integer. It is only used to transfer information about the stress type via the SetValue/GetValue-functions. Principally traced_stress_type is an enum. I also tried to have a Variable of this enum-type but it didn't worked so In decided to use the casting instead.

Reply:
I also would be happy if I could abandon the casting.